doc/smart-switch/reboot/reboot-hld.md

```diff
@@ -0,0 +1,260 @@
+the DPU to terminate all services.
+
+* Upon dispatching the Reboot API, the NPU issues the RebootStatus API to monitor whether the DPU has terminated all services except GNMI and database
+service, continuing until the timeout is reached. Once the DPU successfully terminates all services, it responds to the RebootStatus API with STATUS_SUCCESS.
+Until the services are terminated gracefully, DPU response RebootStatusResponse with STATUS_RETRIABLE_FAILURE status.
+
+* Subsequently, the NPU detaches the DPU PCI. Detachment can be achieved either by a vendor specific API or via sysfs
+(echo 1 > /sys/bus/pci/devices/XXXX:XX:XX.X/remove).
+
+* Next, the NPU triggers a platform vendor API to initiate the reboot process for the DPU.
+
+* The NPU either immediately rescans the PCI upon return or after a timeout period. Rescan of the PCI could be achieved either by calling vendor specific
+API or via sysfs (echo 1 > /sys/bus/pci/devices/XXXX:XX:XX.X/rescan).
+
+## Switch reboot sequence ##
+
+<p align="center"><img src="./images/smartswitch-reboot-seq.svg"></p>
+
+The following outlines the reboot procedure for the entire Smart Switch:
+
+* When the NPU receives a reboot command via the CLI to restart the SmartSwitch, it initiates the reboot sequence.
+
+* The NPU sends a GNOI Reboot API signal to all connected DPUs. This signal instructs the DPUs to gracefully terminate all services, excluding the GNMI
+server, in preparation for the reboot.
+
+* Upon dispatching the Reboot API, the NPU issues the RebootStatus API to monitor whether the DPU has terminated all services except GNMI and database
+service, continuing until the timeout is reached. Once the DPU successfully terminates all services, it responds to the RebootStatus API with STATUS_SUCCESS.
+Until the services are terminated gracefully, DPU response RebootStatusResponse with STATUS_RETRIABLE_FAILURE status.
+
+* Following the confirmation from the DPUs, the NPU proceeds to detach the PCI devices associated with the DPUs. This detachment is achieved either by calling
+vendor specific API or by issuing a command through the sysfs interface, specifically by echoing '1' to the /sys/bus/pci/devices/XXXX:XX:XX.X/remove file
+for each DPU.
+
+* With the DPUs prepared for reboot, the NPU triggers a platform vendor API to initiate the reboot process for the DPUs.
+
+* After initiating the reboot process for the DPUs, the NPU proceeds to reboot itself to complete the overall reboot procedure.
+
+* Upon successful reboot, the NPU resumes operation. As part of the post-reboot process, the NPU may choose to rescan the PCI devices. This rescan operation,
+performed either by invoking vendor API or by echoing '1' to the /sys/bus/pci/devices/XXXX:XX:XX.X/rescan file, ensures that all PCI devices are properly
+recognized and initialized.
+
+## High-Level Design ##
+
+### ModuleBase Class API ###
+
```

Comment:
Do we have any reference for ModuleBase Class API? Is this the vendor API mentioned in the reboot sequence described above?

Reply:
Updated the same. Thank you!